services/web/server/src/simcore_service_webserver/login/handlers.py

```diff
@@ -30,19 +31,24 @@
 REGISTRATION, RESET_PASSWORD, CHANGE_EMAIL = [getattr(ConfirmationAction, att).name
                                 for att in 'REGISTRATION RESET_PASSWORD CHANGE_EMAIL'.split()]
 
-# Handlers & tails ------------------------------------------------------
 
 async def register(request: web.Request):
     _, _, body = await extract_and_validate(request)
 
     # see https://aiohttp.readthedocs.io/en/stable/web_advanced.html#data-sharing-aka-no-singletons-please
+    app_cfg = get_login_config(request.app) # TODO: replace cfg by app_cfg
     db = get_storage(request.app)
+
     email = body.email
-    username = email.split('@')[0]
+    username = email.split('@')[0] # FIXME: this has to be unique and add this in user registration!
```

Comment:
why unique? the name of the user??

Reply:
This is not human-readable email. It is like in github. The username is a short name used as identifier and unique in the system (e.g. in your case there is a unique ``sanderegg`` in github so I can e.g. refer to you in markdown as ``@sanderegg``)